kaldb/pom.xml

```diff
@@ -451,6 +452,45 @@
             </extension>
         </extensions>
         <plugins>
+            <plugin>
+                <groupId>de.thetaphi</groupId>
+                <artifactId>forbiddenapis</artifactId>
```

Comment:
Is there a reason to prefer forbiddenapis library over a the more general checkstyle library?

Reply:
I think this library is configurable so we can add our own rules.

For example, I added a rule saying any `LOG.error` without a stack trace is banned - in our JSON logging setup we rely on the stack trace and index it onto a separate field . 

There was an instance while debugging prod restarts , the log was missing a stack trace. With this rule we'll fix that

Also I am pretty sure our current setup of google error prone is broken - there were talks to add this sort of feature to google error prone but it was never implemented - https://github.com/google/error-prone/issues/1211